python/lsst/eotest/sensor/ptcTask.py

```diff
@@ -167,25 +168,54 @@ def _fit_curves(self, ptc_stats, sensor_id):
         output = EOTestResults(outfile, namps=len(ptc_stats))
         for amp in ptc_stats:
             mean, var = np.array(ptc_stats[amp][0]), np.array(ptc_stats[amp][1])
-            # Compute the median gain and remove outliers.
-            med_gain = np.median(mean/var)
-            frac_resids = np.abs((var - mean/med_gain)/var)
-            index = np.where(frac_resids < self.config.max_frac_offset)
-            # Least-squares fit to the brighter-fatter model.
-            p0 = 0, med_gain
+            index_old = []
+            index = list(np.where(mean < 4e4)[0])
+            count = 1
+            # Initial guess for BF coeff, gain, and square of the read noise
+            pars = 2.7e-6, 0.75, 25
             try:
-                results = scipy.optimize.leastsq(residuals, p0, full_output=1,
-                                                 args=(mean[index], var[index]))
-                pars, cov = results[:2]
+                while index != index_old and count < 10:
+                    results = scipy.optimize.leastsq(residuals, pars, full_output=1,
+                                                     args=(mean[index], var[index]))
+                    pars, cov = results[:2]
+                    sig_resids = residuals(pars, mean, var)
+                    index_old = deepcopy(index)
+                    index = list(np.where(np.abs(sig_resids) < sig_cut)[0])
+                    count += 1
+
+                ptc_a00 = pars[0]
+                ptc_a00_error = np.sqrt(cov[0][0])
                 ptc_gain = pars[1]
                 ptc_error = np.sqrt(cov[1][1])
-            except StandardError as eobj:
+                ptc_noise = np.sqrt(pars[2])
+                ptc_noise_error = 0.5/ptc_noise*np.sqrt(cov[2][2])
+                # Cannot assume that the mean values are sorted
+                ptc_turnoff = max(mean[index])
+            except Exception as eobj:
                 self.log.info("Exception caught while fitting PTC:")
```

Comment:
I'd say that the issue should be uncommon, but clearly it can happen.  The problem with that particular data set is that variances are all zero for two of the flux levels because the two images in the flat pair are identical, which should never happen.  I've added a check to exclude points with zero variance, and a try-except block to catch TypeError and print an explanatory line (on each iteration).  The fit parameters will be bad, but the script will keep running and exit the loop after ten times through.